src/DbConnection.cpp

```diff
@@ -43,6 +43,33 @@ void DbConnection::connect(const cpp11::sexp& host, const cpp11::sexp& user,
     mysql_options(this->pConn_, MYSQL_READ_DEFAULT_FILE,
                   cpp11::as_cpp<std::string>(default_file).c_str());
 
+  // Set SSL options
+  if (client_flag & CLIENT_SSL) {
+#if MYSQL_VERSION_ID >= 80000 && MYSQL_VERSION_ID < 100000
+    // MySQL 8.0 deprecated the enforce SSL flag in prefernce to defining an SSL MODE
+    unsigned int ssl_mode_ = SSL_MODE_REQUIRED;
+    mysql_options(this->pConn_, MYSQL_OPT_SSL_MODE, (void *)&ssl_mode_);
```

Comment:
Is the cast to `void *` necessary here? Omitting it feels slightly clearer, but no strong preference.